api/processors/actions/group/discard.py

```diff
@@ -0,0 +1,62 @@
+from django.db import transaction
+
+from api.models import Group, Teacher
+from api.models.auxil.status_setter import StatusSetter
+from api.models.choices.log_event_type import StudentLogEventType, TeacherLogEventType
+from api.models.choices.status import StudentProjectStatus, TeacherProjectStatus
+from api.processors.actions.group import GroupActionProcessor
+from api.processors.auxil.log_event_creator import GroupLogEventCreator
+
+
+class GroupDiscardProcessor(GroupActionProcessor):
+    def __init__(self, group: Group, reason: str):
+        self.reason = reason
+        super().__init__(group)
+
+    @transaction.atomic
+    def process(self) -> None:
+        self._set_statuses()
+        self._create_log_events()
+        self._delete()
+
+    def _create_log_events(self) -> None:
+        GroupLogEventCreator.create(
+            group=self.group,
+            student_log_event_type=StudentLogEventType.TENTATIVE_GROUP_DISCARDED,
+            teacher_log_event_type=TeacherLogEventType.TENTATIVE_GROUP_DISCARDED,
+            comment=self.reason,
+        )
+
+    def _delete(self) -> None:
+        self.group.delete()
+
+    def _set_group_status(self) -> None:
+        super()._set_group_status()
```

Comment:
Do we need to redefine this method if it only calls `super()`?

Reply:
In general yes, because it is an abstract method and you have to define it but we might just leave pass but I prefer to call super because in some cases it might contain default implementation

Reply:
Ah, okay